catalog/spatial/wfs/spatial-wfs-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/catalog/source/reader/XmlSchemaMessageBodyReader.java

```diff
@@ -66,20 +81,33 @@ public XmlSchema readFrom(
       MultivaluedMap<String, String> headers,
       InputStream inStream)
       throws IOException, WebApplicationException {
-    // Determine if this is an XMLSchema
-    String input = IOUtils.toString(inStream);
+
+    String input = IOUtils.toString(inStream, StandardCharsets.UTF_8);
     inStream.reset();
-    String count = COUNT_XPATH_BUILDER.evaluate(camelContext, input);
-    // See if there exactly one instance of "xsd:schema" in this doc
-    if (Integer.parseInt(count) == 1) {
-      XmlSchema schema = null;
+
+    if (isValid(new InputSource(inStream))) {
```

Comment:
Does an InputSource need to be closed?

Reply:
No, an InputSource is not closable. It just reads from the underlying stream. As for whether or not to close the stream, the javadoc for MessageBodyReader says not to [1]:
> The caller is responsible for ensuring that the input stream ends when the entity has been consumed. The implementation should not close the input stream.

[1] https://eclipse-ee4j.github.io/jaxrs-api/apidocs/2.1.6/javax/ws/rs/ext/MessageBodyReader.html#readFrom(java.lang.Class,java.lang.reflect.Type,java.lang.annotation.Annotation%5B%5D,javax.ws.rs.core.MediaType,javax.ws.rs.core.MultivaluedMap,java.io.InputStream)